text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,1443 @@
+this category easier.
+
+[nom]: https://crates.io/crates/nom
+[peresil]: https://crates.io/crates/peresil
+
+This helps address the goal of "Rust should provide easy access to high quality
+crates" as stated in the [Rust 2017 Roadmap][roadmap].
+
+[roadmap]: https://github.com/rust-lang/rfcs/pull/1774
+
+# Detailed design
+[design]: #detailed-design
+
+Please see the [Appendix: Comparative Research][comparative-research] section
+for ways that other package manager websites have solved this problem, and the
+[Appendix: User Research][user-research] section for results of a user research
+survey we did on how people evaluate crates by hand today.
+
+A few assumptions we made:
+
+- Measures that can be made automatically are preferred over measures that
+  would need administrators, curators, or the community to spend time on
+  manually.
+- Measures that can be made for any crate regardless of that crate's choice of
+  version control, repository host, or CI service are preferred over measures
+  that would only be available or would be more easily available with git,
+  GitHub, Travis, and Appveyor. Our thinking is that when this additional
+  information is available, it would be better to display a badge indicating it
+  since this is valuable information, but it should not influence the ranking
+  of the crates.
+- There are some measures, like "suitability for the current task" or "whether
+  I like the way the crate is implemented" that crates.io shouldn't even
+  attempt to assess, since those could potentially differ across situations for
+  the same person looking for a crate.
+- We assume we will be able to calculate these in a reasonable amount of time
+  either on-demand or by a background job initiated on crate publish and saved
+  in the database as appropriate. We think the measures we have proposed can be
+  done without impacting the performance of either publishing or browsing
+  crates noticeably. If this does not turn out to be the case, we will have to
+  adjust the formula.
+
+## Factors
+
+Through [the survey we conducted][user-research], we found that when people
+evaluate crates, they are looking primarily for approximate signals of:
+
+- Ease of use
+- Maintenance
+- Quality
+
+Feeding those signals are related measures of:
+
+- Popularity
+- Credibility
+
+We detail how we propose to address each of these in turn, plus a rating of the
+five crates from the user research survey as examples.
+
+We'd like to provide a coarse binning of the scores in each category, to avoid
+over-analyzing the difference between, say, 72% and 78% and seeing significance
+where there isn't really one. We've considered using letter grades, but those
+often have emotional associations (F means you're a failure), when it should be
+just an indicator of reality and not a value judgment. So we're also proposing
+an option of an emoji scale and are open to other proposals:
```

Comment:
Seems like a 5-star system would be more easily recognizable and fit the same level of coarseness.